apps/client/src/app/publish/publish.component.ts

```diff
@@ -45,19 +49,47 @@ export class PublishComponent implements OnInit {
     private implementationGuideService: ImplementationGuideService,
     private cookieService: CookieService,
     public configService: ConfigService,
-    private exportService: ExportService) {
+    private exportService: ExportService,
+    private http: HttpClient) {
 
     this.options.implementationGuideId = !this.route.snapshot.paramMap.get('id') ?
       this.cookieService.get(Globals.cookieKeys.exportLastImplementationGuideId + '_' + this.configService.fhirServer) :
       this.route.snapshot.paramMap.get('id');
     this.options.responseFormat = <any>this.cookieService.get(Globals.cookieKeys.lastResponseFormat) || 'application/json';
-
+    this.options.template = <any>this.cookieService.get(Globals.cookieKeys.lastTemplate) || this.options.template;
+    this.options.templateVersion = <any>this.cookieService.get(Globals.cookieKeys.lastTemplateVersion) || this.options.templateVersion;
     // Handle intermittent disconnects mid-export by notifying the server that we are currently exporting the given packageId
     this.socketService.onConnected.subscribe(() => {
       if (this.packageId) {
         this.socketService.notifyExporting(this.packageId);
       }
     });
+
+    this.templateChanged
```

Comment:
it might be worth considering putting this duplicate/reused logic in one of the services on the client.
Perhaps consider `ConfigService.getTemplateVersions(templateName: string)`.
Then you could re-use the ConfigService to get this information in both the ExportComponent and PublishComponent.